src/shapes/Object/InteractiveObject.ts

```diff
@@ -436,13 +436,14 @@ export class InteractiveFabricObject<
           this.height,
           calcDimensionsMatrix(options)
         ),
-        stroke = (
-          this.strokeUniform
-            ? new Point().scalarAdd(this.canvas ? this.canvas.getZoom() : 1)
-            : // this is extremely confusing. options comes from the upper function
-              // and is the qrDecompose of a matrix that takes in account zoom too
-              new Point(options.scaleX, options.scaleY)
-        ).scalarMultiply(this.strokeWidth);
+        stroke = !this.isStrokeAccountedForInDimensions()
+          ? (this.strokeUniform
+              ? new Point().scalarAdd(this.canvas ? this.canvas.getZoom() : 1)
+              : // this is extremely confusing. options comes from the upper function
+                // and is the qrDecompose of a matrix that takes in account zoom too
+                new Point(options.scaleX, options.scaleY)
+            ).scalarMultiply(this.strokeWidth)
+          : ZERO;
```

Comment:
this is bad practice, having custom methods that repeatedly calculate the same thing differently.
We should IMO use aCoords and render in the scene plane. That way we do not need to handle so many cases and we can stick to fixing aCoords once instead f finding these issues again and again